project/SparkBuild.scala

```diff
@@ -1928,3 +1928,25 @@ object TestSettings {
   )
 
 }
+
+object DependencyVersions {
+  import java.util.Properties
+
+  private lazy val properties = settingKey[Properties]("Effective POM properties")
+  lazy val jacksonVersion = settingKey[String]("Jackson version")
+  lazy val jacksonModuleID = settingKey[ModuleID]("Jackson Module ID")
+  lazy val settings = Seq(
+    properties := SbtPomKeys.effectivePom.value.getProperties,
+    jacksonVersion := getVersion(properties.value, "fasterxml.jackson"),
+    jacksonModuleID := ModuleID("com.fasterxml.jackson", "jackson-bom", jacksonVersion.value),
+  )
+
+  private def getVersionFromPom(properties: Properties, property: String) : String = {
+    properties.get(property).asInstanceOf[String]
+  }
+
+  private def getVersion(properties: Properties, property: String) : String = {
+    val version = property + ".version"
+    sys.props.get(version).getOrElse(getVersionFromPom(properties, version))
+  }
```

Comment:
small suggestion, hope to make it clearer

```suggestion
  private lazy val pomProps = settingKey[Properties]("Effective POM properties")
  lazy val jacksonVersion = settingKey[String]("Jackson version")
  lazy val jacksonBom = settingKey[ModuleID]("Jackson BOM Module ID")
  lazy val settings = Seq(
    pomProps := SbtPomKeys.effectivePom.value.getProperties,
    jacksonVersion := getProperty(pomProps.value, "fasterxml.jackson.version"),
    jacksonBom := ModuleID("com.fasterxml.jackson", "jackson-bom", jacksonVersion.value),
  )

  private def getProperty(pomProps: Properties, property: String) : String = {
    sys.props.get(property).getOrElse(pomProps.get(property).asInstanceOf[String])
  }
```

BTW, could you please move `object DependencyVersions` to line 1180?